securing_apps/topics/oidc/keycloak-gatekeeper.adoc

```diff
@@ -0,0 +1,494 @@
+
+==== HTTP routing
+
+By default all requests will be proxyed on to the upstream, if you wish to ensure all requests are authentication you can use this:
+
+[source,bash]
+----
+--resource=uri=/* # note, unless specified the method is assumed to be 'any|ANY'
+----
+
+The HTTP routing rules follow the guidelines from link:https://github.com/go-chi/chi#router-design[chi]. The ordering of the resources do not matter, the router will handle that for you.
+
+==== Session-only cookies
+
+By default the access and refresh cookies are session-only and disposed of on browser close; you can disable this feature using the `--enable-session-cookies` option.
+
+==== Forward-signing proxy
+
+Forward-signing provides a mechanism for authentication and authorization between services using tokens issued from the IdP. When operating in this mode the proxy will automatically acquire an access token (handling the refreshing or logins on your behalf) and tag outbound requests with a Authorization header. You can control which domains are tagged with the --forwarding-domains option. Note, this option use a **contains** comparison on domains. So, if you wanted to match all domains under *.svc.cluster.local you can use: --forwarding-domain=svc.cluster.local.
+
+At present the service performs a login using oauth client_credentials grant type, so your IdP service must support direct (username/password) logins.
+
+Example setup:
+
+You have collection of micro-services which are permitted to speak to one another; you have already set up the credentials, roles, and clients in Keycloak, providing granular role controls over issue tokens.
+
+[source,yaml]
+----
+- name: {generic_adapter_name}
+  image: quay.io/gambol99/keycloak-generic-adapter:latest
+  args:
+  - --enable-forwarding=true
+  - --forwarding-username=projecta
+  - --forwarding-password=some_password
+  - --forwarding-domains=projecta.svc.cluster.local
+  - --forwarding-domains=projectb.svc.cluster.local
+  - --tls-ca-certificate=/etc/secrets/ca.pem
+  - --tls-ca-key=/etc/secrets/ca-key.pem
+  # Note: if you don't specify any forwarding domains, all domains will be signed; Also the code checks is the
+  # domain 'contains' the value (it's not a regex) so if you wanted to sign all requests to svc.cluster.local, just use
+  # svc.cluster.local
+  volumeMounts:
+  - name: keycloak-socket
+    mountPoint: /var/run/keycloak
+- name: projecta
+  image: some_images
+
+# test the forward proxy
+$ curl -k --proxy http://127.0.0.1:3000 https://test.projesta.svc.cluster.local
+----
+
+On the receiver side you could set up the {generic_adapter_name_full} (--no=redirects=true) and permit this to verify and handle admission for you. Alternatively, the access token can found as a bearer token in the request.
```

Comment:
Maybe the correct is `can be found` instead of `can found`?

Reply:
Good catch!